### PR TITLE
Test/unit integration test react

### LINKED ISSUE #40: test/react-user-component
As a dev,

I would like to have all User elements tested

(Unit Testing)

### LINKED ISSUE #41: test/react-navigation-testing
As a dev, 

I would like to see Navigation elements and functions tested

(Unit testing)

### LINKED ISSUE #42: test/react-single-event-component
As a dev,

I would like to see PostingView elements and functions tested

(Unit testing)
(Integration test) for subtracting open positions

### LINKED ISSUE #43: test/react-posting-card
As a dev,

I would like to see PostingCards elements and functions tested

(Unit testing)

### LINKED ISSUE #44: test/react-postings
As a dev,

I would like to see Postings elements and functions tested

(Unit testing)

### LINKED ISSUE #45: test/react-app
As a dev,

I would like to see App elements and functions tested

(Unit testing)
(Integration testing:
-click on a positing card to redirect to a single page
-click on a upcoming job to redirect to a single page
-sort, filter, and search display the correct events on click and change
-signing up for the job changes the list of the upcoming jobs on the user sidebar
-clicking on a return button sends user to a home page "/"
)

### PR DESCRIPTION
**Pull request by okalimbet**
#### What's this PR do?
Create Unit Test for every component:
* User
* PostingView
* Postings
* PostingCard
* Navigation
* App

Create Integration Test for:
* App
* PostingView 

Installed Dependencies:
* testing-library/react
* testing-library/jest-dom
* testing-library/user-event 

#### What does it fix?
N/A

#### Where should the reviewer start?
App.test.js line 1
User.test.js line 1
Postings.test.js line 1
PostingCard.test.js line 1
Navigation.test.js line 1
Navigatoin/index.js - line 27
PostingCard/index.js line 9
PostingView/index.js  line 14
Postings/index.js line 25
User/index.js line 21
_mockData.js line 1

#### How should this be manually tested?
By running `npm test` to pass all the tests and running `npm start` to display fully functional website

### CODE REVIEW

File: src/Components/App/App.test.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+describe("App", () => {
+
+  beforeEach(() => {
+    apiCalls.getUser.mockResolvedValue(_mockData.users[0]);
+    apiCalls.getPostings.mockResolvedValue(_mockData.events);
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks();
```
Comment (coopterrones) [inline src/Components/App/App.test.js:21]: This was huge...

File: src/Components/App/App.test.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  it("App should be rendered on load and display loading before all data is fetched", async() => {
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    )
+
+    expect(screen.getByText("LOADIN\'...")).toBeInTheDocument()
+
+    await act(() => Promise.resolve());
```
Comment (coopterrones) [inline src/Components/App/App.test.js:33]: have not used act before... I will have to catch up on what this is about. Will look into it.

File: src/Components/App/App.test.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    )
+    await waitFor(() => expect(apiCalls.getUser).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(apiCalls.getPostings).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(screen.getByTestId("navigation-element")).toBeInTheDocument());
+
+    const filteredCategory = screen.getByTestId("select-input");
+    fireEvent.change(filteredCategory, {
```
Comment (yiranli624) [inline src/Components/App/App.test.js:179]: It still puzzles me why this one interference with the single page redirect test...

**Review by yiranli624 — APPROVED** [review]
Great job us! Great work on one whole day testing!!

File: src/Components/App/App.test.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    apiCalls.getUser.mockResolvedValue(_mockData.updatedUser[0]);
+
+    const history = createMemoryHistory();
+    render(
+      <Router history={history}>
+          <App />
+        </Router>
+      )
+      
+    const mockEvent = await waitFor(() => screen.getAllByTestId("posting-card-element"));
```
Comment (coopterrones) [inline src/Components/App/App.test.js:112]: funny naming convention here.. I have questions...

File: src/Components/App/App.test.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    await waitFor(() => expect(apiCalls.getPostings).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(screen.getByTestId("navigation-element")).toBeInTheDocument());
+
+    const userInput = screen.getByPlaceholderText("i.e Boulder...");
+    userEvent.type(userInput, 'Food Delivery');
+
+    const searchbutton = screen.getByText("search");
+    fireEvent.click(searchbutton);
+
+    await waitFor(() => expect(screen.getByText("Food Delivery")).toBeInTheDocument())
```
Comment (coopterrones) [inline src/Components/App/App.test.js:142]: looks good

File: src/Components/Navigation/Navigation.test.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+describe("Navigation", () => {
+
+  let categories, 
+      searchByKeyWord,
+      filterByCategory,
+      sortByDate;
+
+  beforeEach(() => {
+    categories = mockData.events.reduce((totalCategories, posting) => {
```
Comment (coopterrones) [inline src/Components/Navigation/Navigation.test.js:17]: Smart !

**Pull request merged**